### PR TITLE
Match Tensorflows implementation of GoogLeNet more closely

### PR DESCRIPTION
**Pull request by TheCodez**
Given that we only provide pretrained weights converted from TensorFlow it makes sense to just resemble their version.
Their implementation also doesn't have aux branches so it's probably better to just remove them as we don't have pretrained weights for them and having them seems to cause a lot of confusion for some users.

### CODE REVIEW

**Comment by soumith** [comment]
from what I understand, it doesn't train without the aux branches. They released the weights for the main branch, but they train with the aux branch too.

**Comment by TheCodez** [comment]
@soumith maybe that was only the case in the original paper implementation without batch normalization. The TensorFlow implementation hoewever uses batch normalization so there's a chance they were able to train it without the aux branches.

Anyways we could either:
- Remove the aux branches
- Add a doc telling users that we don't have pretrained weights for them and/or disable them when using the `pretrained` option.

**Comment by fmassa** [comment]
I would say that GoogleNet should be treated as an inference-only model. We've not managed to train it after a few trials, so maybe just overriding `.train` to avoid setting the model to `train` mode could be a solution?

**Comment by soumith** [comment]
okay yea removing aux branches and making it training only seems to make sense

**Comment by TheCodez** [comment]
I disabled the branches when using pretrained and added a comment explaining the parameter. This still allows one to use those branches if desired.

@fmassa @soumith what do you guys think?

File: torchvision/models/googlenet.py [hunk 1]
```diff
@@ -56,25 +60,25 @@ def __init__(self, num_classes=1000, aux_logits=True, transform_input=False, ini
...
-        if aux_logits:
-            self.aux1 = InceptionAux(512, num_classes)
-            self.aux2 = InceptionAux(528, num_classes)
+
+        self.aux1 = InceptionAux(512, num_classes)
+        self.aux2 = InceptionAux(528, num_classes)
+
         self.avgpool = nn.AdaptiveAvgPool2d((1, 1))
-        self.dropout = nn.Dropout(0.4)
+        self.dropout = nn.Dropout(0.2)
```
Comment (fmassa) [inline torchvision/models/googlenet.py]: I suppose the value of dropout was wrong before? Can you expand on this?

File: torchvision/models/googlenet.py [hunk 2]
```diff
@@ -56,25 +60,25 @@ def __init__(self, num_classes=1000, aux_logits=True, transform_input=False, ini
...
-                nn.init.xavier_uniform_(m.weight)
-                if m.bias is not None:
-                    nn.init.constant_(m.bias, 0.2)
-            elif isinstance(m, nn.Linear):
-                nn.init.xavier_uniform_(m.weight)
-                nn.init.constant_(m.bias, 0)
+            if isinstance(m, nn.Conv2d) or isinstance(m, nn.Linear):
+                import scipy.stats as stats
+                X = stats.truncnorm(-2, 2, scale=0.01)
+                values = torch.Tensor(X.rvs(m.weight.numel()))
```
Comment (fmassa) [inline torchvision/models/googlenet.py]: Can you use `torch.as_tensor` instead, by passing `dtype` to match the dtype of the weight? `torch.Tensor` as a constructor is legacy now

File: torchvision/models/googlenet.py [hunk 3]
```diff
@@ -56,25 +60,25 @@ def __init__(self, num_classes=1000, aux_logits=True, transform_input=False, ini
...
+                values = values.view(m.weight.size())
+                m.weight.data.copy_(values)
```
Comment (fmassa) [inline torchvision/models/googlenet.py]: I think that it's better to use
```python
with torch.no_grad():
    m.weight.copy_(values)
```
here, as `.data` is legacy as well

File: torchvision/models/googlenet.py [hunk 4]
```diff
@@ -56,25 +60,25 @@ def __init__(self, num_classes=1000, aux_logits=True, transform_input=False, ini
 
         self.inception5a = Inception(832, 256, 160, 320, 32, 128, 128)
         self.inception5b = Inception(832, 384, 192, 384, 48, 128, 128)
-        if aux_logits:
-            self.aux1 = InceptionAux(512, num_classes)
-            self.aux2 = InceptionAux(528, num_classes)
+
+        self.aux1 = InceptionAux(512, num_classes)
```
Comment (fmassa) [inline torchvision/models/googlenet.py]: While I'm ok with this change (specially that the default value of `aux_logits` was true), this is a BC-breaking change because previously saved state_dicts won't work out of the box anymore if the user had saved a dict with `aux_logits=False`.

I think a solution would be to modify `load_state_dict` for this model to handle this.

@soumith do you think this is necessary?

File: torchvision/models/googlenet.py (hunk 1 above)
Comment (TheCodez) [reply torchvision/models/googlenet.py]: The paper states 0.4 while TensorFlow uses 0.2. They probably decreased dropout because they added batch normalization.

File: torchvision/models/googlenet.py (hunk 3 above)
Comment (TheCodez) [reply torchvision/models/googlenet.py]: I can also send a pr to fix the code for inception v3

File: torchvision/models/googlenet.py (hunk 4 above)
Comment (TheCodez) [reply torchvision/models/googlenet.py]: There isnt't a official release that includes GoogLeNet yet, so this shouldn't be much of a concern?

Users could still load their model by passing `strict=False` to `load_state_dict` I think.

Comment (fmassa) [reply torchvision/models/googlenet.py]: Actually you are right, googlenet was added after 0.2.2 was released, so this is fine
